src/base/bittorrent/session.cpp

```diff
@@ -127,6 +127,47 @@ namespace
         return result;
     }
 
+    template <typename Entry>
+    QSet<QString> entryListToSetImpl(const Entry &entry)
+    {
+        Q_ASSERT(entry.type() == Entry::list_t);
+        QSet<QString> output;
+        for (int i = 0; i < entry.list_size(); ++i) {
+            const QString tag = QString::fromStdString(entry.list_string_value_at(i));
+            const bool valid = Session::isValidTag(tag);
+            if (valid)
+                output.insert(tag);
+            else
+                qWarning() << QString("Dropping invalid stored tag: %1").arg(tag);
+        }
+        return output;
+    }
+
+#if LIBTORRENT_VERSION_NUM < 10100
+    bool isList(const libt::lazy_entry *entry)
+    {
+        return entry && entry->type() == libt::lazy_entry::list_t;
+    }
+
+    QSet<QString> entryListToSet(const libt::lazy_entry *entry)
+    {
+        if (entry)
```

Comment:
Done, but I had to replace the braces with a direct call to the container constructor, because the compiler cannot deduce the correct type from braces in a conditional.